mapper/orm-coordination-outbox-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/outboxpolling/event/impl/OutboxPollingOutboxEventSendingPlan.java

```diff
@@ -54,7 +55,7 @@ public void discard() {
 
 	@Override
 	public <R> CompletableFuture<MultiEntityOperationExecutionReport<R>> sendAndReport(
-			EntityReferenceFactory<R> entityReferenceFactory) {
+			EntityReferenceFactory<R> entityReferenceFactory, OperationSubmitter ignore) {
```

Comment:
Unless you have a good reason to think this is irrelevant, I would at least assert that the submitter is `BLOCKING` here (and through an `AssertionFailedException` if not).

SPI contracts are no joke, you can't just ignore them like that :)

Reply:
I think you forgot to throw an exception?

There are two possibilities:

1. either this method is always non-blocking (that's what I meant by "Unless you have a good reason to think this is irrelevant") and you don't need to assert anything (not even to log a warning): the method will always be compatible with either submitter.
2. or this method is always blocking and you need to fail hard when the assertion fails, because we're not complying with the requested submitter.

As far as I can tell we're calling `persist` at some point, so this method is most likely blocking. So I would fail hard.